utility_analysis_new/parameter_tuning.py

```diff
@@ -67,12 +67,19 @@ class TuneOptions:
           if this argument is a callable, it should take 1 argument of type
           AggregateErrorMetrics and return float.
         parameters_to_tune: specifies which parameters to tune.
+        partitions_sampling_prob: the probability with which each partition
```

Comment:
From the description it looks like this method might potentially apply sampling. We discussed that we might also want to apply sampling at pre-aggregation stage. 
Do you think it might be helpful to mention that the caller might apply sampling before passing the input dataset to this function and what kind of sampling can be used. Or is it too generic / obvious?